docs/framework/dataframe.mdx

```diff
@@ -2,97 +2,213 @@
+`record()` returns a row in an `EditableDataFrame`. It takes an integer specifying the index of the row.
+
+In the code example above, you would retrieve the record at row 1 with the following code:
 
 ```python
-import pandas
-import writer as wf
+record = state["mydf"].record(1)
+```
 
-df = pandas.DataFrame({'a': [1, 2, 3], 'b': [4, 5, 6]})
-wf.init_state({
-	'mydf': wf.EditableDataframe(df)
-})
+### `record_update`: Change an existing row
 
-# Subscribe this event handler to the `wf-dataframe-add` event
-def on_record_add(state, payload):
-	payload['record']['sales'] = 0 # default value inside the dataframe
-	state['mydf'].record_add(payload)
+`record_update()` replaces an existing row in an `EditableDataFrame` with a new one. It takes a dictionary with the following structure...
 
+```python
+{
+	"record_index": index,
+	"record": row_to_update
+}
+```
 
-# Subscribe this event handler to the `wf-dataframe-update` event
-def on_record_change(state, payload):
-    state['mydf'].record_update(payload)
+...where `index` is an integer specifying which row should be updated and `row_to_update` is a dictionary containing the updated row data.
 
+In the code example above, you would update the row at index 0 with the following code:
 
-# Subscribe this event handler to the `wf-dataframe-action` event
-def on_record_action(state, payload):
-	"""
-	This event corresponds to a quick action in the drop-down menu to the left of the dataframe.
-	"""
-	record_index = payload['record_index']
-	if payload['action'] == 'remove':
-		state['mydf'].record_remove(payload)
-	if payload['action'] == 'open':
-		state['record'] = state['df'].record(record_index) # dict representation of record
+```python
+state["mydf"].record_update({
+	"record_index": 0,
+	"record": {"rank": 6, "flavor": "Bubble gum", "favorite": 0.08}
+})
 ```
 
-### Datastructures supported by `EditableDataframe`
+### `record_remove`: Delete an existing row
 
-`EditableDataframe` can be used with a panda dataframe, a polar dataframe and list of records.
+`record_remove()` removes an existing row from an `EditableDataFrame`. It takes a dictionary with the following structure...
 
 ```python
-import pandas
-import polars
+{"record_index": index}
+```
 
-import writer as wf
+...where `index` is an integer specifying which row should be deleted.
 
-panda_df = pandas.DataFrame({'a': [1, 2, 3], 'b': [4, 5, 6]})
-polars_df = polars.DataFrame({'a': [1, 2, 3], 'b': [4, 5, 6]})
-list_of_records = [{'a': 1, 'b': 4}, {'a': 2, 'b': 5}, {'a': 3, 'b': 6}]
+In the code example above, you would delete the row at index 2 with the following code:
 
-wf.init_state({
-	'mypandas': wf.EditableDataframe(panda_df),
-	'mypolars': wf.EditableDataframe(polars_df),
-	'mylistofrecords': wf.EditableDataframe(list_of_records)
-})
-```
+```python
+state["mydf"].record_remove({"record_index": 2})
```

Comment:
Seems like this is missing a conclusion or something to wrap this up and tell the user where to go next